views/Hangouts.tsx

```diff
@@ -75,7 +82,18 @@ class Hangouts extends React.Component<Props> {
       this.props.ongoingHangouts.length > 0,
     profileVisible: Object.keys(this.props.currentProfile).length > 0,
     visibleHangout:
-      this.props.receivedHangoutRequests.length > 0 ? 'pending' : 'accepted'
+      this.props.receivedHangoutRequests.length > 0 ? 'pending' : 'accepted',
+    active: this.props.isActive
+  }
+
+  setUserLocation(bool) {
+    const updatedUser = {
+      longitude: this.props.longitude,
+      latitude: this.props.latitude,
+      is_active: bool
+    }
+    this.setState({ active: bool })
```

Comment:
We need to use the boolean from the state instead of the store, as that will give the toggle switch a smooth animation. The is_active bool in the store in set asynchronously and will make the switch toggle 2 times